src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/TestHelper/NotificationHubsManagementHelper.cs

```diff
@@ -0,0 +1,191 @@
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace NotificationHubs.Tests.TestHelper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using Microsoft.Azure;
+    using Microsoft.Azure.Management.NotificationHubs;
+    using Microsoft.Azure.Management.NotificationHubs.Models;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Management.Resources.Models;
+    using Microsoft.Azure.Test;
+    using Microsoft.Azure.Test.HttpRecorder;
+    using Microsoft.IdentityModel.Clients.ActiveDirectory;
+    using Microsoft.WindowsAzure.Management;
+    using System.Security.Cryptography;
+
+    public static class NotificationHubsManagementHelper
+    {
+        internal const string DefaultLocation = "South Central US";
+        internal const string ResourceGroupPrefix = "NotificationHub-RG";
+        internal const string NamespacePrefix = "HydraNH-Namespace";
+        internal const string NotificationHubPrefix = "HydraNH-NotificationHub";
+        internal const string AuthorizationRulesPrefix = "HydraNH-Authrules";
+        internal const string DefaultNamespaceAuthorizationRule = "RootManageSharedAccessKey";
+
+        public static NotificationHubsManagementClient GetNotificationHubsManagementClient()
+        {
+            return TestBase.GetServiceClient<NotificationHubsManagementClient>(new CSMTestEnvironmentFactory());
+        }
+
+        public static ResourceManagementClient GetResourceManagementClient()
+        {
+            return TestBase.GetServiceClient<ResourceManagementClient>(new CSMTestEnvironmentFactory());
+        }
+
+        public static ManagementClient GetManagementClient()
+        {
+            return TestBase.GetServiceClient<ManagementClient>();
+        }
+
+        public static void RefreshAccessToken(this NotificationHubsManagementClient notificationHubsManagementClient)
+        {
+            if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                // if it's playback then do nothing
+                return;
+            }
+
+            var testEnvironment = new CSMTestEnvironmentFactory().GetTestEnvironment();
+            var context = new AuthenticationContext(new Uri(testEnvironment.Endpoints.AADAuthUri, testEnvironment.Tenant).AbsoluteUri);
+
+            var result = context.AcquireToken("https://management.core.windows.net/", testEnvironment.ClientId, new Uri("urn:ietf:wg:oauth:2.0:oob"), PromptBehavior.Auto);
```

Comment:
Please do it like this: https://github.com/Azure/azure-sdk-for-net/blob/master/src/ResourceManagement/Storage/Storage.Tests/Helpers/StorageManagementTestUtilities.cs#L63